src/components/CorporationsFilter.ts

```
@@ -1,67 +1,44 @@
 import Vue from "vue";
 
 import { CardName } from "../CardName";
-import { CorporationGroup } from "../CorporationName";
-import { ALL_CORPORATION_DECKS } from "../cards/AllCards";
-import { PRELUDE_CARD_MANIFEST } from "../cards/prelude/PreludeCardManifest";
-import { VENUS_CARD_MANIFEST } from "../cards/venusNext/VenusCardManifest";
-import { COLONIES_CARD_MANIFEST } from "../cards/colonies/ColoniesCardManifest";
-import { CORP_ERA_CARD_MANIFEST, BASE_CARD_MANIFEST } from "../cards/StandardCardManifests";
-import { PROMO_CARD_MANIFEST } from "../cards/promo/PromoCardManifest";
-import { TURMOIL_CARD_MANIFEST } from "../cards/turmoil/TurmoilCardManifest";
-import { CardManifest } from "../cards/CardManifest";
-import { COMMUNITY_CARD_MANIFEST } from "../cards/community/CommunityCardManifest";
+import { ALL_CARD_MANIFESTS, ALL_CORPORATION_DECKS, MANIFEST_BY_MODULE } from "../cards/AllCards";
+import { GameModule } from "../GameModule";
 
+function corpCardNames(module: GameModule): Array<CardName> {
+    return MANIFEST_BY_MODULE.get(module)!.corporationCards.cards.map(cf => cf.cardName);
```

Using the `!` operator is dangerous. If `module` is not found in the map this will throw a runtime exception. If `MANIFEST_BY_MODULE.get(module) === undefined` this can return `[]`.

---

Yeah, I saw you having issues with ! before, but here me out. MANIFEST_BY_MODULE is a map keyed by GameModule, so under what conditions could this actually occur? That's why I think it's safe here. My personal experience is that returning [] under what is a guaranteed impossible occurrence means that when it happens it hides the error.

If you disagree and you have a reasonable rebuttal, let me know. I'm also not going to fight for it, but it's important you know my reasons -- a few more of these! are going to come up with Ares (particularly inside AresHandler -- but more on that another time.)

---

I see that you marked it for all of them -- feel free to not have to deeply respond to each one as if I asked them independently -- feel free to respond to all of them right here, or something.